src/controllers/hid/hidenumerator.cpp

```diff
@@ -105,3 +117,22 @@ QList<Controller*> HidEnumerator::queryDevices() {
 
     return m_devices;
 }
+
+bool HidEnumerator::isDuplicate(hid_device_info *dev)
+{
+    QHash <QString,QString > info;
+    HidController *hid;
+    int i;
+
+    info["vendor_id"] = QString::number(dev->vendor_id, 16);
+    info["product_id"] = QString::number(dev->product_id, 16);
+
+    for (i = 0; i < m_devices.count(); i++) {
```

Comment:
you can use a `foreach` macro here.

```C++
foreach(const HidController * dev, m_devices) {
    if (dev->matchProductionInfo(info)) {
         return true;
    }
}
````

I'm not 100% sure though that the cast will work here as well in the macro. It could likely be that you have to to the cast in the loop.